src/App.js

```diff
@@ -14,11 +14,129 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isGoodsShown: false,
+    isReversed: false,
+    lengthFilter: 1,
+  }
+
+  showGoods = () => {
+    this.setState({
+      isGoodsShown: true,
+    });
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+      lengthFilter: 1,
+    });
+  }
+
+  reverseGoods = () => {
+    this.setState(({ isReversed }) => ({
+      isReversed: !isReversed,
+    }));
+  }
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortBy: 'alphabet',
+    });
+  }
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  }
+
+  render() {
+    const {
+      isGoodsShown,
+      isReversed,
+      sortBy,
+      lengthFilter,
+    } = this.state;
+    const visibleGoods = goodsFromServer.filter(
+      good => good.length >= lengthFilter,
+    );
+
+    visibleGoods.sort((curGood, nextGood) => {
+      switch (sortBy) {
+        case 'alphabet':
+          return curGood.localeCompare(nextGood);
+        case 'length':
+          return curGood.length - nextGood.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    if (isGoodsShown) {
+      return (
+        <div className="App">
+          <h1>Goods</h1>
+          <p>
+            Set filter by word length:
+            <select
+              value={lengthFilter}
+              onChange={({ target }) => {
+                this.setState({
+                  lengthFilter: target.value,
+                });
+              }}
+            >
+              <option>1</option>
```

Comment:
Let's not write so many lines of code, but use an array of numbers from 1 to 10 and map `options` from it